runtime/runtime.go

```diff
@@ -108,6 +116,27 @@ func (r *Runtime) GetInstanceAttributes(ctx context.Context, instanceID string)
 	return instanceAnnotationsToAttribs(instance)
 }
 
+// cleanTempDirs removes temporary directories of every instance.
+// This is to prevent temp data from accumulating in a persistent directory over time due to bugs or crashes.
+// In happy cases any temp files should be cleared by the logic that adds them.
+func cleanTempDirs(dataDir string, logger *zap.Logger) {
+	fsRoot := os.DirFS(dataDir)
+	glob := path.Clean(path.Join("./", filepath.Join("*", "tmp")))
+
+	matches, err := doublestar.Glob(fsRoot, glob)
+	if err != nil {
+		logger.Warn("failed to list temp directories", zap.Error(err))
+		return
+	}
+
+	for _, match := range matches {
+		err := os.RemoveAll(match)
+		if err != nil {
+			logger.Warn("failed to remove temp directory", zap.String("dir", match), zap.Error(err))
+		}
+	}
+}
```

Comment:
It might be cleaner to do this in `registryCache.add` (i.e. before creating the empty `tmp` directory, delete any existing `tmp` directory).

Reply:
Yeah I also thought so but actually that may not handle all failures cases like during `DeleteInstance instance was deleted from registry but controller did not stop and server crashed. 
Since this is an extra check just to ensure we handle all failure cases as best as possible, I felt this is the best place to do that.